fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/ServerPlayerEvents.java

```diff
@@ -23,11 +23,31 @@
 import net.fabricmc.fabric.api.event.EventFactory;
 
 public final class ServerPlayerEvents {
+	/**
+	 * An event that is called before a player is spawned when opening a world or joining a server.
+	 *
+	 * <p>Mods may use this event to set the player's world and position before it is completely spawned.
+	 *
+	 * <p>Note: this event is not called when a player is respawned.
+	 * For that purpose, {@link ServerPlayerEvents#COPY_FROM} should be used.
+	 */
+	public static final Event<BeforeSpawn> BEFORE_SPAWN = EventFactory.createArrayBacked(BeforeSpawn.class, callbacks -> (player) -> {
```

Comment:
BEFORE_JOIN_SERVER maybe a better name? Also dont want to confuse it with ServerPlayConnectionEvents.JOIN

If anyone has any suggestions please let me know.

Reply:
BEFORE_FIRST_SPAWN is also an option because I think it's about the first spawn in a World?